### PR TITLE
Enable config updater plugin

### LINKED ISSUE #41: Automatic update of official Prow after merge to master branch
After merging to the test-infra **master** branch, official Prow cluster should be automatically updated.

See https://github.com/kubernetes/test-infra/tree/master/prow/plugins/updateconfig

### PR DESCRIPTION
**Pull request by crabtree**
**Description**

Changes proposed in this pull request:

- Enable config updater plugin for test-infra repository
- Introduce copy of starter.yaml file for prow installation on a cluster
- Introduce config.yaml and plugins.yaml generated for cluster
- Fix hook role for fetching config maps
- Remove unused configuration from config.yaml.tpl
- Documentation update

**Related issue(s)**

Resolves #41

### CODE REVIEW

File: prow/cluster/starter.yaml [hunk 1]
```diff
@@ -0,0 +1,543 @@
+---
```
Comment (aszecowka) [inline prow/cluster/starter.yaml]: I would like to know, from which revision this file is copied.
Additionally, please document how this file differs from a original one.

File: prow/cluster/starter.yaml [hunk 2]
```diff
@@ -0,0 +1,543 @@
...
+apiVersion: v1
+kind: ConfigMap
```
Comment (aszecowka) [inline prow/cluster/starter.yaml]: install-prow.sh should now apply this file, not from ```https://raw.githubusercontent.com/kubernetes/test-infra/a202e595a33ac92ab503f913f2d710efabd3de21/prow/cluster/starter.yaml```

Comment (crabtree) [reply prow/cluster/starter.yaml]: Done

File: prow/cluster/starter.yaml (hunk 1 above)
Comment (crabtree) [reply prow/cluster/starter.yaml]: Done

**Pull request merged**